appIcons.js

```diff
@@ -747,6 +741,32 @@ const MyAppIcon = new Lang.Class({
         return backgroundColor;
     },
 
+    /**
+     * Downsample large icons before scanning for the backlight color to
+     * improve performance.
+     *
+     * @param pixBuf
+     * @param pixels
+     * @param resampleX
+     * @param resampleY
+     *
+     * @return [];
+     */
+    _resamplePixels: function (pixels, pixBuf, resampleX, resampleY) {
```

Comment:
Thanks for the refactoring!

Small bug: the `pixels` and `piBuf` inputs are swapped.
```
_resamplePixels: function (pixels, pixBuf, resampleX, resampleY)
_resamplePixels: function (pixBuf, pixels, resampleX, resampleY)
```